libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1384,14 +1444,11 @@ MQTTStatus_t MQTT_SerializeConnect( const MQTTConnectInfo_t * pConnectInfo,
                     pBuffer ) );
         status = MQTTBadParameter;
     }
-    /* Check that the full packet size fits within the given buffer. */
-    else if( connectPacketSize > pBuffer->size )
+    /* A buffer must be configured for serialization. */
+    else if( pBuffer->pBuffer == NULL )
```

Comment:
This naming `pBuffer->pBuffer` is confusing.  Consider renaming the outer `pBuffer`.

Reply:
To be consistent I would have to rename it everywhere, but I will do it here for now.